plugins/websearch/src/lib.rs

```diff
@@ -0,0 +1,115 @@
+use abi_stable::std_types::{ROption, RString, RVec};
+use anyrun_plugin::*;
+use serde::Deserialize;
+use std::{fmt, fs, process::Command};
+use strum::IntoEnumIterator;
+use strum_macros::EnumIter;
+use urlencoding::encode;
+
+#[derive(Debug, Clone, Copy, Deserialize, EnumIter)]
+enum Engine {
+    Google,
+    Ecosia,
+    Bing,
+    DuckDuckGo,
+}
+
+impl Engine {
+    fn value(&self) -> &str {
+        match *self {
+            Engine::Google => "google.com/search?q=",
```

Comment:
These URLs should probably contain the `https://` prefix.

Reply:
the protocol is defined globally for all engines in line 103 `xdg-open https://{}{}`

Reply:
Oh nevermind me then, I'm just blind.